src/parse_error.rs

```diff
@@ -1,19 +1,37 @@
 use alloc::string::String;
 
+#[cfg(not(feature = "std"))]
+use core::fmt;
+#[cfg(feature = "std")]
+use std::fmt;
```

Comment:
This can be simplified to just `use core::fmt;`, which works both with and without `no_std`.